src/transformers/hf_argparser.py

```diff
@@ -145,6 +150,15 @@ def _add_dataclass_arguments(self, dtype: DataClassType):
                     kwargs["required"] = True
             parser.add_argument(field_name, **kwargs)
 
+            # Add a complement `no_*` argument for a boolean field AFTER the initial field has already been added.
+            # Order is important for arguments with the same destination!
+            # We use a copy of earlier kwargs because the original kwargs have changed a lot before reaching down
+            # here and we do not need those changes/additional keys.
+            if field.default is True and (field.type is bool or field.type == Optional[bool]):
```

Comment:
Should we do `bool_kwargs["default'] = False` just in case the key is already in the kwargs?

Reply:
Ah, yes that is probably a better solution!